test/specs/fetch.specs.ts

```diff
@@ -0,0 +1,63 @@
+import { expect } from "chai";
+import JSONBig from "json-bigint";
+
+import { APIError } from "../../src/errors/APIError";
+import { checkStatus, parseJSON } from "../../src/utils/fetch";
+
+const createResponse = (data: Record<string, unknown>, status = 200) =>
+    new Response(JSONBig({ useNativeBigInt: true }).stringify(data, null, 2), {
+        "status": status,
+        "statusText": "Dummy status",
+    });
+
+describe("Fetch Helpers", () => {
+    describe("parseJSON", () => {
+        it("return the parsed response body", async () => {
+            const data = {
+                key1: "Value 1",
+                key2: { key21: "Value 2" },
+                key3: 1234567890,
+                key4: true,
+                key5: { key51: "Value 51" },
+                key6: [{ key61: "Value 61" }, 12345, false],
+            };
+
+            const result = await parseJSON(createResponse(data));
+
+            expect(result).eql({
+                key1: "Value 1",
+                key2: { key21: "Value 2" },
+                key3: 1234567890,
+                key4: true,
+                key5: { key51: "Value 51" },
+                key6: [{ key61: "Value 61" }, 12345, false],
+            });
+        });
+
+        it("return the parsed response body without formatting the keys in parameter", async () => {
+            const result = await parseJSON(createResponse({ key1: "1", key2: "2", key3: { key_31: 1 } }), ["key3"]);
+
+            expect(result).eql({ key1: "1", key2: "2", key3: { key_31: 1 } });
+        });
+
+        it("return the parsed response body with bigint", async () => {
+            const result = await parseJSON(createResponse({ key1: BigInt("0123456789101112131415") }), ["key3"]);
```

Comment:
`123456789101112131415n` maybe looks better?

Reply:
Indeed, a lot nicer, thanks! 🙂

Reply:
Hum... Seems that node 12 does not really like the target being es2020 and ts wouldn't allow me to choose something lower that that. Since this is a minor change we may want to have it in another PR.